retry.go

```diff
@@ -114,10 +114,14 @@ func ExponentialBackoff(backoff Backoff, logger logger, condition ConditionFunc)
 
 type auroraThriftCall func() (resp *aurora.Response, err error)
 
+// Verification functions should strive to keep the number of thrift calls to one wherever possible.
+type verifyOnTimeout func() (*aurora.Response, bool)
```

Comment:
Does it make sense to pass to this function the response and error that was returned by the earlier thrift call? Some verify function may need it.

For someone who does not know the code well, it would be very helpful to add comment to explain the expected behavior of this function -- what is it supposed to verify, that does the bool return argument mean, where is the aurora response supposed to come from

Reply:
In this case it the response and error don't give us any information that we don't already have. By the time the verification function is called, we've already made sure we're in a very specific timeout scenario where the response is invalid due to the error not being nil and the error has already been dissected.

This feature, as a whole, became a compromise. 

I have long leaned on keeping this sort of mechanism outside of gorealis because I feel we're headed towards bloat. I've argued that the caller should be responsible for this type of activity.

I finally relented after weighing the pros and cons of including a mechanism where, when a timeout from the client side is experienced, and it is unknown if the thrift call made it to the scheduler, a rudimentary check is automatically done to determine if the call made it successfully to the scheduler or not.

Will add some more comments to explain the function signature.